modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -63,7 +63,8 @@ sealed class UnwrappedCrossSources(
     *   CrossSources with all the scripts wrapped
     */
   def withWrappedScripts(buildOptions: BuildOptions): CrossSources = {
-    val codeWrapper = ScriptPreprocessor.getScriptWrapper(buildOptions)
+    val sharedOptions = this.sharedOptions(buildOptions)
```

Comment:
So shared options should contain the right Scala version?

Reply:
Yes, when this method is passed the buildOptions created from CLI options it will merge them with the buildOptions obtained from using directives, thus we'll get the final version of options here.

Reply:
Awesome, thanks!